### ISSUE TITLE
Configuration generator

### ISSUE THREAD

**Issue by L3tum** [netdata/netdata#5125]
I don't know how others feel about this, who have been using this awesome program for a long time now, but as a newbie getting into it the configuration is just way too overwhelming. 
I don't care about half the stats netdata is collecting by default, so I have to turn that off (by searching through the entire sample configuration to try to find a relevant sounding configuration entry), but I do have around 10 or so apps I want to monitor that I then need to figure out how to properly configure, where the config files are and whether it would tank my system or not. 

I understand it's probably not as high on the priority list, if at all, but right now I'd probably spend 2+ hours just configuring netdata without any finetuning. 

##### Feature idea summary

So in my opinion, a generator asking basic questions and writing that into a config file would be a good idea. It can be more or less detailed, since most of the finetuning wouldn't make sense in a generator. But questions like "update interval" or "disable/enable these internal/well-known plugins" or "are you using this program?" would be really good. 

##### Expected behavior

Basically a CLI or whatever where you get asked basic questions and the results are written into a netdata.conf file, maybe with a notice of other config files generated (for individual plugins) or at least references to the ones you'd need.

**Comment by cakrit** [comment]
We've been discussing this need recently and we definitely want to do a lot of work in this area. We have a lot of config options and we need to create some wizards to simplify the process. It won't be a CLI, but a GUI.  

Thanks for creating the issue, I've been meaning to create one myself.

**Comment by paulfantom** [comment]
If we want to do it via GUI, then we would need to store config in `/var/lib/netdata` instead of `/etc/netdata` as files in `/etc` shouldn't be modified by a software reading those files. Also configuration via GUI doesn't really go hand-in-hand with any provisioning tool (ansible, chef, puppet, etc.)

On the other hand separate CLI application could read and modify `/etc/netdata` to create meaningful configuration. Basically a better, more interactive `edit-config` script. Maybe written in a way which uses `dialog` command to create a nice TUI.

**Comment by L3tum** [comment]
@cakrit I'm glad to hear that!

I agree with @paulfantom however. Everyone probably knows these "self-installers" shipped by Symfony, Angular and the like which ask some basic questions to fine-tune the installation a bit and are IMO really useful.
If you go down the GUI route it'd need to be some standalone thing, like a simple website akin to [Ninite](https://ninite.com/). Downloading an application to a GUI system (my server is not, for example), only to then copy the result back to the target system seems counter-intuitive in my opinion. A website would eliminate the first step.
Although I wouldn't use a config generator with provisioning tools but rather use it to get a base config set up which I can then either use directly or fine-tune further if I want/need to, either through provisioning tools or manually.

**Comment by cakrit** [comment]
Just so we don't go off track at this early stage, let me say that the general idea for the first stage is something like the following:
- Build the wizards on a website, which will almost certainly be the netdata cloud UI we're building. This will enable generating and later applying configuration changes to multiple agents.
- The UI will communicate with the netdata agents, via the existing API, to retrieve information like netdata.conf, charts etc. It will need this info to make the wizard smart. The API will need an extension to return more config files than just netdata.conf.
- The first version will certainly just be producing only a few config files or file parts that the admin can download and use as s/he chooses. For example, one big quick win is to build a wizard for new alarms/alarm templates and just return a proper configuration for one alarm/alarm template. In later versions, we could offer the option to apply the config changes to the netdata agents.

**Comment by Ferroin** [comment]
For the long-term goal of being able to apply the config to individual Netdata agents, I would suggest looking at how BOINC handles computing preferences.  They let you configure a central set of preferences, with some basic grouping functionality (you can group systems, and then apply a set of preferences to that group), but each individual system can locally override individual directives within the main configuration.  This approach makes it easy to manage configuration at scale, while still offering the option of local overrides when they are needed.

I think for Netdata we might want to offer a choice to the user to decide which config (local or remote) takes precedence when they define different values for the same option.  Probably should be controlled by the network config, and probably default to local overriding network.

**Comment by cakrit** [comment]
The long-term goal is related to UC1 and UC3 of #4731

**Comment by ilyam8** [comment]
Closing this feature request. We will re-evaluate the request internally.
